# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/frontend/ParserAtom.h
# Commit: cb0e58368cd4
# Full Hash: cb0e58368cd4d8e1f79b193b7ccae9586cc2ab50
# Author: Tooru Fujisawa <arai_a@mac.com>
# Date: 2020-11-14 09:46:25
# Regressor Bug: 1675241
# File Overlap Count: 1
# Description:
#   Bug 1675241 - Part 10: Use TaggedParserAtomIndex in StencilModuleEntry. r=tcampbell
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D96917
# ==============================================================================

diff -r 73cc5fa0e0bd -r cb0e58368cd4 js/src/frontend/ParserAtom.h
--- a/js/src/frontend/ParserAtom.h	Sat Nov 14 00:03:20 2020 +0000
+++ b/js/src/frontend/ParserAtom.h	Sat Nov 14 00:03:28 2020 +0000
@@ -117,6 +117,11 @@
   explicit constexpr TaggedParserAtomIndex(StaticParserString2 index)
       : data_(uint32_t(index) | WellKnownTag | Static2SubTag) {}
 
+  static TaggedParserAtomIndex star() {
+    return TaggedParserAtomIndex(StaticParserString1('*'));
+  }
+  static TaggedParserAtomIndex null() { return TaggedParserAtomIndex(); }
+
   bool isParserAtomIndex() const {
     return (data_ & TagMask) == ParserAtomIndexTag;
   }